nipype/interfaces/afni/preprocess.py

```diff
@@ -285,21 +291,23 @@ def _format_arg(self, name, trait_spec, value):
         return super(Allineate, self)._format_arg(name, trait_spec, value)
 
     def _list_outputs(self):
-        outputs = self.output_spec().get()
+        outputs = self._outputs().get()
         if not isdefined(self.inputs.out_file):
-            outputs['out_file'] = self._gen_filename(self.inputs.in_file,
-                                                     suffix=self.inputs.suffix)
+            outputs['out_file'] = self._gen_fname(self.inputs.in_file,
+                                                  suffix='_allineate.nii')
```

Comment:
@rmarkello - thanks for this clarification and sorry for the delay in responding to this. perhaps we can do this in two stages. first lets fix this interface. then we can use a separate PR to implement the logic you outlined above.